microsetta_admin/server.py

```diff
@@ -774,6 +775,217 @@ def scan():
                                       recorded_type)
 
 
+def _get_color_code():
+    val = '0123456789ABCDEF'
+    code = ["#"+''.join([random.choice(val) for i in range(6)])]
+    return code[0]
+
+
+def _get_legends(criteria):
+
+    dict = {}
+    if criteria not in session:
+        status, fields = APIRequest.get(
+            '/api/admin/barcode_query_fields')
+
+        if status == 200:
+            for obj in fields:
+                if criteria == "Type" and obj["label"] == "Sample Site":
+                    for val in obj["values"]:
+                        color_code = _get_color_code()
+                        while color_code in dict:
+                            color_code = _get_color_code()
+                        dict[obj["values"][val]] = color_code
+
+                if criteria == "Status" and obj["label"] == "Sample Status":
+                    for val in obj["values"]:
+                        color_code = _get_color_code()
+                        while color_code in dict:
+                            color_code = _get_color_code()
+                        tmp = obj["values"][val].replace(' ', '-').lower()
+                        dict[tmp] = color_code
+
+                if criteria == "Project" and obj["label"] == "Project":
+                    for val in obj["values"]:
+                        if not ('-' in obj["values"][val]):
+                            color_code = _get_color_code()
+                            while color_code in dict:
+                                color_code = _get_color_code()
+                            dict[obj["values"][val]] = color_code
+
+            if len(dict) > 0:
+                dict['None'] = "#000000"
+                session[criteria] = dict
+    else:
+        dict = session.get(criteria)
+
+    return dict
+
+
+def _compose_table(legends, type):
+    output_dict = {}
+    scanned_data = session.get('scan_data')
+    current_row = ""
+    current_cols = []
+
+    for rec in scanned_data:
+        status, data = APIRequest.get(
+            '/api/admin/search/samples/%s' % rec[5]
+        )
+
+        val = ""
+        proj = data["sample"]["sample_projects"][0]
+        status = data["sample"]["_latest_scan_status"]
+        src = data["sample"]["site"]
+
+        if type == "Type":
+            val = data["sample"]["site"]
+        elif type == "Status":
+            val = data["sample"]["_latest_scan_status"]
+        elif type == "Project":
+            val = data["sample"]["sample_projects"][0]
+
```

Comment:
In case of invalid type, function returns all scanned samples associated with "None" type implying that none of the samples match the criteria selected for visualization.